### PR TITLE
feat: serve frontend from backend

### PR DESCRIPTION
**Pull request by G3root**
# Description 📣

fixes: #1058

- [x] New feature
- [x] Breaking change

# Tests 🛠️

```sh
# Here's some code block to paste some code snippets
```

---

- [ ] I have read the [contributing guide](https://infisical.com/docs/contributing/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/code-of-conduct). 📝

### CODE REVIEW

**Comment by maidul98** [comment]
I see you still have pm2 and nginx in the docker image. I think we should be able to remove nginx because all the traffic should go right to the backend now and after that we'll have one process so we can remove pm2 as well. What do you think?

**Comment by maidul98** [comment]
> I see you still have pm2 and nginx in the docker image. I think we should be able to remove nginx because all the traffic should go right to the backend now and after that we'll have one process so we can remove pm2 as well. What do you think?

I have addressed these. New issue is fixing issue with getting data fetched via `getStaticProps`.

Screenshot 2023-10-16 at 12 50 27 AM

https://github.com/Infisical/infisical/blob/3f0b6dc6c1d9b51fbb4014f059919e3b5e546c51/frontend/src/pages/integrations/%5Bid%5D.tsx#L28

Second issue: something is up with rendering integrations page

**Comment by G3root** [comment]
> > I see you still have pm2 and nginx in the docker image. I think we should be able to remove nginx because all the traffic should go right to the backend now and after that we'll have one process so we can remove pm2 as well. What do you think?
> 
> I have addressed these. New issue is fixing issue with getting data fetched via `getStaticProps`.
> Screenshot 2023-10-16 at 12 50 27 AM
> 
> https://github.com/Infisical/infisical/blob/3f0b6dc6c1d9b51fbb4014f059919e3b5e546c51/frontend/src/pages/integrations/%5Bid%5D.tsx#L28
> 
> Second issue: something is up with rendering integrations page

pushed a fix for that

File: .dockerignore [hunk 1]
```diff
@@ -1,2 +1,11 @@
...
+frontend/node_modules
+backend/frontend-build
+**/node_modules
+**/.next
+.dockerignore
+.git
+README.md
+.dockerignore
+**/Dockerfile
+.next
```
Comment (Salman2301) [inline .dockerignore:11]: We can remove this?

Comment (Salman2301) [reply .dockerignore:11]: Already have `**/.next`

**Review by maidul98 — APPROVED** [review]
app

**Pull request merged**